add covered_locs output to coverage

clarkbw opened the pull request:
Hello 👋 

I've been working with the @codecov team on the flow coverage report so I can upload my flow coverage report to http://codecov.io/ and use the service to analyze coverage over time.

One thing I was told was missing from the flow coverage report (in json format) was a `covered_locs` output.  There is `uncovered_locs` but no inverse for `covered_locs`, only the sum `covered_count` which makes it difficult for them to create the line-by-line report. Because the flow report only shows lines not covered Codecov never knows when lines are covered so it always results in 0% coverage. 🐼 

This change simply adds `covered_locs` as an additional field to the json output.  I can't foresee that this could be a breaking change for anyone ingesting the json output. It does create slightly larger output files as all covered lines now need to be described as well.

I've only just given the latest coverage output to Codecov, when I hear back I'll update this if there are any additional issues but I wanted to get this conversation started in the mean time.

stevepeak commented:
> @clarkbw awesome job 👍 

Hey I'm from the @codecov team and integrated Flow reports (given @clarkbw changes) into Codecov. I'm happy to be the point of contact if there are any questions/comments. Cheers!

clarkbw commented:
Maybe @samwgoldman or @aackerman could a look?

aackerman commented:
@clarkbw I'm not on the flow team, I just happened to have touched this code last. It looks fine to me anyways. 👍

clarkbw commented:
> I just happened to have touched this code last. It looks fine to me anyways

Ok, thanks @aackerman!

clarkbw commented:
Woo!

clarkbw commented:
Anything I can do to help here?

DanielMSchmidt commented:
What is the state of this?

JustinC474 commented:
Hi Flow Team! Was wondering if we could get an update on the state of this? We're trying to add flow coverage to our codecov reporting and so it'd be hugely helpful to have this merged.

Also, thanks @clarkbw for putting this together!

cc\ @mroch @int3

digitalmaster commented:
Any movement on this?

AndresCuervo commented:
Any word on this PR? Seems like it was pulled into Phabricator, would be great if someone with access to that could respond or at least give an update if any further work needs to be done?

clarkbw commented:
Rebased and updated this PR to resolve the conflict that was created when f6ae79145356dd8f0e1e1b1b985365721cd42426 was merged a couple weeks go.  😫 😢 😭

JustinC474 commented:
Thanks @clarkbw! This PR would be awesome to get in, appreciate you keeping it up to date so that its ready to go when Facebook gets around to it.

tmc commented:
@clarkbw can you address the ci failure?
@mroch this tiny patch has been sitting here for nearly a year.

clarkbw commented:
It’s an unrelated intermittent failure. Happy to rebase again but I don’t think there’s a point unless this might actually get merged.

tmc commented:
@clarkbw might as well remove that barrier.

Peeja commented:
Anything remaining that's holding this back? I'd really like to start using it.

MoOx commented:
This would be really nice to have better coverage utilities to block PR etc if coverage goes down. I believe this PR is the least we can accept to start moving forward! Diff is REALLY small!

AndresCuervo commented:
+1 to the above comments asking for this small change to be merged. The way I see it:

- it's a small change & appears to the solve the problem with no apparent drawbacks (in this case, ready to merge, would be great if someone with collaborator access could get approval & merge!)
- if there are concerns/conflicts with something at Facebook it would be great if those could be communicated, if there are considerations/alterations that could be made to the PR so someone (presumably @clarkbw) could make those updates & we can get it merged soon.

Thanks!

vicapow commented:
cc @__avik this would really help with improving flow line coverage reports. also see: https://github.com/rpl/flow-coverage-report/issues/67

avikchaudhuri commented:
Merging this, sorry for it falling off the radar.

vicapow commented:
Woo! Thanks, @avikchaudhuri
